tests/test_security_path_traversal.py

```
@@ -204,3 +204,77 @@ def test_run_tailer_path_validation(tmp_path):
+        with pytest.raises(HTTPException) as exc_info:
+            await validate_evolution_patch_endpoint("../etc", "patch_1")
+        assert exc_info.value.status_code == 400
+
+        with pytest.raises(HTTPException) as exc_info:
+            await validate_evolution_patch_endpoint("run_1", "..\\windows")
+        assert exc_info.value.status_code == 400
+
+        # apply_evolution_patch_endpoint
+        with pytest.raises(HTTPException) as exc_info:
+            req = ApplyEvolutionRequest(patch_id="../etc", dry_run=True, create_backup=True)
+            await apply_evolution_patch_endpoint(req)
```

Ruff (0.15.2)

[warning] `pytest.raises()` block should contain a single simple statement

(PT012)